### PR TITLE
2389 as a player i wish to be able to delete orders

### LINKED ISSUE #2389: As a player I wish to be able to delete orders
I've entered a set of orders, but now I wish to cancel them.

Note: since it's not easy to add an `Archive` button into JSON Editor, I've changed the approach. Now umpire staff are able to archive orders documents.

The functionality is implemented in #2419 , but the implementation of `archiveConfirmed` needs to be finished off, to send the modified messages to the database.

To reflect the "time-travelling" aspect of Serge, we won't delete the document, we'll mark it as cancelled, using the existing `archived` flag in `MessageDetails`.  The client-side will not display cancelled orders. Note: we can't just strip out the cancelled versions of orders, since we would then just display the previous version of the orders.  We need to filter the orders in `PlanningChannel`.

### PR DESCRIPTION
**Pull request by lilitkarapetyan**
Fixes #2389

### CODE REVIEW

File: packages/client/src/api/wargames_api/index.ts [hunk 1]
```diff
@@ -700,6 +700,11 @@ const checkReference = (message: MessageCustom, db: ApiWargameDb, details: Messa
   })
 }
 
+export const PostBulkMessages = (dbName: string, archiveMark: MessagePlanning[]) => {
```
Comment (IanMayo) [inline packages/client/src/api/wargames_api/index.ts]: it looks like there is a bit to complete here too, @lilitkarapetyan  🤣

Comment (lilitkarapetyan) [reply packages/client/src/api/wargames_api/index.ts:703]: I am workin on it 🤣🤣

Comment (lilitkarapetyan) [reply packages/client/src/api/wargames_api/index.ts]: I am working on it  🤣🤣

**Comment by lilitkarapetyan** [comment]
I haven't finished.

File: packages/components/src/local/organisms/planning-messages-list/index.tsx [hunk 2]
```diff
@@ -218,6 +218,8 @@ export const PlanningMessagesList: React.FC<PropTypes> = ({
         return msg
       })
       console.log('Archiving:', markArchived)
+      // if (markArchived.length === 0) return null
```
Comment (IanMayo) [inline packages/components/src/local/organisms/planning-messages-list/index.tsx:259]: @lilitkarapetyan - this isn't working, but it's in my code, not yours :-(

At this point in the code, the `pendingArchive` array is empty.  I've inserted debug lines to capture the array getting set (in `archiveSelected`), but in `archiveConfirmed` the array is empty.  React State...

I'd appreciate you helping out with this issue, please.

Comment (lilitkarapetyan) [reply packages/components/src/local/organisms/planning-messages-list/index.tsx:259]: I tested there with `rows` data,

**Comment by IanMayo** [comment]
Hello @lilitkarapetyan - I've had a play with this branch.  

I know you are currently still developing it.

I see the new rows are being added to the database, marked as `archived`.  But, they still have the original `_id`. 

The player UI isn't updating.  That may be because of the `special` way we decide to show new messages in `PlanningMessagesList` (to prevent forms data-loss).  If that is an issue, we can temporarily overcome it by asking players to `browser refresh`,  knowing it will be fixed in #2452 

Hmm, this is interesting :-).  The list of messages _does_ update for other players - just not the one doing the `archive`. Now that's strange :-)

**Comment by lilitkarapetyan** [comment]
> Hello @lilitkarapetyan - I've had a play with this branch.
> 
> I know you are currently still developing it.
> 
> I see the new rows are being added to the database, marked as `archived`. But, they still have the original `_id`.
> 
> The player UI isn't updating. That may be because of the `special` way we decide to show new messages in `PlanningMessagesList` (to prevent forms data-loss). If that is an issue, we can temporarily overcome it by asking players to `browser refresh`, knowing it will be fixed in #2452
> 
> Hmm, this is interesting :-). The list of messages _does_ update for other players - just not the one doing the `archive`. Now that's strange :-)

Do you wanr archived messages to have no original _id ?

**Comment by lilitkarapetyan** [comment]
We can pass a rendom Id and add a new message to the database.

**Comment by lilitkarapetyan** [comment]
Yes, the update problem comes from the how we update out table

**Comment by IanMayo** [comment]
Aah, of course.

Yes, in the current implementation, the messages go back with their original ID, which makes them replace the original one.  Hmm, that's not the normal Serge way of doing things.  I did try deleting the `_id` - hoping Serge would create one, but that didn't work (the messages didn't get inserted).  

So, yes - let's provide a new ID.  I _think_ it has to be a moment.utc().toISOtimestamp().  

Note: I tried doing this in `archiiveConfirmed` in `PlanningMessagesList` - but it didn't work.  We (you) will have to set the new `_id` later in the process, please .

**Comment by lilitkarapetyan** [comment]
> 

Okay, Thanks

**Comment by IanMayo** [comment]
aah, here is the issue I was trying to overcome.  Do you have another solution?
image

**Comment by lilitkarapetyan** [comment]
> 

No, I will study.

**Comment by IanMayo** [comment]
> > 
> 
> No, I will study.

@lilitkarapetyan - if there isn't a quick/easy fix, please wrap the code in comments that disable the linter check.  We've got lots of other things to work on.

**Comment by IanMayo** [comment]
@lilitkarapetyan  - is this ready to go?

**Pull request merged**